server/pages/index.js

```diff
@@ -128,6 +128,13 @@ function getDefaultContext( request ) {
 		initialServerState = getInitialServerState( serializeCachedServerState );
 	}
 
+	const protocol = request.get( 'X-Forwarded-Proto' ) === 'https' ? 'https' : 'http';
```

Comment:
I can use that method to get the protocol if you want, I just wanted to point out that we already build up a very similar URL elsewhere in the same file.  Would seem odd to use two different methods of creating the URL.  If we're going to use `request.protocol` here, I'd suggest using it in the other location as well.